### PR TITLE
IKASAN-1587 - Changed ScheduledConsumer to Consumer interface on Buil…

### PR DESCRIPTION
**Pull request by mitcje**
…der generic.

### CODE REVIEW

File: ikasaneip/builder/jar/src/test/java/org/ikasan/builder/FlowBuilderTest.java [hunk 1]
```diff
@@ -1530,4 +1530,97 @@ public void test_successful_withMonitor_implied_build()
...
+
+    /**
+     * Test successful flow creation.
+     */
+    @Test
+    public void test_successful_consumer_implied_and_explicit_builds()
+    {
+        BuilderFactory builderFactory = ikasanApplication.getBuilderFactory();
+
+        // implied
```
Comment (Chris-McQueen-Development) [inline ikasaneip/builder/jar/src/test/java/org/ikasan/builder/FlowBuilderTest.java]: pedantic point; implicit might be the better word? Directly the opposite of explicit

File: ikasaneip/builder/jar/src/test/java/org/ikasan/builder/FlowBuilderTest.java [hunk 2]
```diff
@@ -1530,4 +1530,97 @@ public void test_successful_withMonitor_implied_build()
...
+                .build();
+
+        // explicit
+        builderFactory.getFlowBuilder("moduleName", "flowName")
+                .consumer("consumerName", builderFactory.getComponentBuilder().dbConsumer().build())
+                .producer("producerName", producer)
+                .build();
+
+//        // implied
+//        builderFactory.getFlowBuilder("moduleName", "flowName")
```
Comment (Chris-McQueen-Development) [inline ikasaneip/builder/jar/src/test/java/org/ikasan/builder/FlowBuilderTest.java]: How comes this is left in commented out?

File: ikasaneip/builder/jar/src/test/java/org/ikasan/builder/FlowBuilderTest.java [hunk 3]
```diff
@@ -1530,4 +1530,97 @@ public void test_successful_withMonitor_implied_build()
...
+                .build();
+
+        // explicit
+        builderFactory.getFlowBuilder("moduleName", "flowName")
+                .consumer("consumerName", builderFactory.getComponentBuilder().jmsConsumer().build())
+                .producer("producerName", producer)
+                .build();
+
+//        // implied
+//        builderFactory.getFlowBuilder("moduleName", "flowName")
```
Comment (Chris-McQueen-Development) [inline ikasaneip/builder/jar/src/test/java/org/ikasan/builder/FlowBuilderTest.java]: How comes this is left in commented out?

File: ikasaneip/builder/jar/src/test/java/org/ikasan/builder/component/endpoint/DbConsumerBuilderTest.java [hunk 4]
```diff
@@ -140,6 +141,8 @@ public void dbConsumer_build_when_all_attributes_set_scheduledConsumer_vanilla_m
                 .setSqlStatement("sqlStatement")
                 .build();
 
+        ScheduledConsumer scheduledConsumer = (ScheduledConsumer)consumer;
```
Comment (Chris-McQueen-Development) [inline ikasaneip/builder/jar/src/test/java/org/ikasan/builder/component/endpoint/DbConsumerBuilderTest.java]: Pedantic point; this:
`...(ScheduledConsumer)consumer`
Should be this:
`...(ScheduledConsumer) consumer`
>

Comment (Chris-McQueen-Development) [reply ikasaneip/builder/jar/src/test/java/org/ikasan/builder/component/endpoint/DbConsumerBuilderTest.java]: Unless that's not our pattern?

File: ikasaneip/builder/jar/src/test/java/org/ikasan/builder/FlowBuilderTest.java (hunk 2 above)
Comment (mitcje) [reply ikasaneip/builder/jar/src/test/java/org/ikasan/builder/FlowBuilderTest.java]: Yep - my bad - this test is failing due to unrelated issues, but will see if I can fix it

File: ikasaneip/builder/jar/src/test/java/org/ikasan/builder/FlowBuilderTest.java (hunk 3 above)
Comment (mitcje) [reply ikasaneip/builder/jar/src/test/java/org/ikasan/builder/FlowBuilderTest.java]: Yep - my bad - this test is failing due to unrelated issues, but will see if I can fix it

**Pull request merged**